### PR TITLE
Initial content

### PR DESCRIPTION
**Pull request by ferricoxide**
Adds baseline functionality to project

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
 # rsyslog-formula
 
 This Salt formula will install and configure the rsyslog system-logging service. It is expected that this automation will be executed as part of a larger SaltStack execution such as that performed when running [Watchmaker](https://watchmaker.readthedocs.io/).
+
+This formula expects that any remote syslog servers' information will be stored in a site-specifica Pillar-configuration. See the pillar.example file for what the Pillar-structure should look like. If nothing appropriate is defined in Pillar, informational-output to that effect will be emitted.
```
Comment (lorengordon) [inline README.md:5]:
```suggestion
This formula expects that any remote syslog servers' information will be stored in a site-specific Pillar-configuration. See the pillar.example file for what the Pillar-structure should look like. If nothing appropriate is defined in Pillar, informational-output to that effect will be emitted.
```

File: rsyslog/subcomponent/config/files/default/subcomponent-example.tmpl
Comment (lorengordon) [inline rsyslog/subcomponent/config/files/default/subcomponent-example.tmpl:1]: Let's just delete this subcomponent directory. It's not relevant to the formula.

File: rsyslog/config/clean.sls [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# vim: ft=sls
+
+{#- Get the `tplroot` from `tpldir` #}
+{%- set tplroot = tpldir.split('/')[0] %}
+{%- set sls_service_clean = tplroot ~ '.service.clean' %}
+{%- from tplroot ~ "/map.jinja" import mapdata as TEMPLATE with context %}
```
Comment (lorengordon) [inline rsyslog/config/clean.sls:7]: Can search/replace the whole project to change `TEMPLATE` to `rsyslog`

File: rsyslog/clean.sls [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# vim: ft=sls
+
+include:
+  - .subcomponent.clean
```
Comment (lorengordon) [inline rsyslog/clean.sls:5]: Can delete the references to `subcomponent`

File: rsyslog/config/file.sls [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+{%- set tplroot = tpldir.split('/')[0] %}
+{%- set sls_package_install = tplroot ~ '.package.install' %}
+{%- from tplroot ~ "/map.jinja" import mapdata as rsyslog with context %}
+{%- from tplroot ~ "/libtofs.jinja" import files_switch with context %}
+{%- set loghost_dict = salt.pillar.get('rsyslog:lookup:remote_hosts', {}) %}
+
+include:
+  - {{ sls_package_install }}
+
+{%- if loghost_dict|length == 0 %}
```
Comment (lorengordon) [inline rsyslog/config/file.sls:14]:
```suggestion
{%- if not loghost_dict %}
```

File: rsyslog/config/file.sls [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  test.show_notification:
+    - text: |
+        ----------------------------------------
+        No Pillar information supplied for
+        configuring logging to a remote rsyslog
+        server
+        ----------------------------------------
+{%- else %}
+  {%- for loghost, values in loghost_dict.items() %}
+    {%- if values.encrypted == True or values.relp == True %}
```
Comment (lorengordon) [inline rsyslog/config/file.sls:25]:
```suggestion
    {%- if values.encrypted or values.relp %}
```

File: rsyslog/config/file.sls [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+{%- from tplroot ~ "/libtofs.jinja" import files_switch with context %}
+{%- set loghost_dict = salt.pillar.get('rsyslog:lookup:remote_hosts', {}) %}
+
+include:
+  - {{ sls_package_install }}
+
+{%- if loghost_dict|length == 0 %}
+No Remote syslog defined:
+  test.show_notification:
+    - text: |
```
Comment (lorengordon) [inline rsyslog/config/file.sls:17]: We can fail this condition. Don't call the formula if not configuring it.

```suggestion
  test.fail_without_changes:
    - comment: |
```

File: rsyslog/config/file.sls [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        No Pillar information supplied for
+        configuring logging to a remote rsyslog
+        server
+        ----------------------------------------
+{%- else %}
+  {%- for loghost, values in loghost_dict.items() %}
+    {%- if values.encrypted == True or values.relp == True %}
+Bad sub-option:
+  test.show_notification:
+    - text: |
```
Comment (lorengordon) [inline rsyslog/config/file.sls:28]: This is also a fail condition:

```suggestion
  test.fail_without_changes:
    - comment: |
```

**Review by lorengordon — DISMISSED** [review]
There's a bit of cleanup to do on the TEMPLATE bits, but overall makes sense

**Comment by lorengordon** [comment]
Will also need the basic Makefile and Dockerfile and .github/dependabot.yaml to get the tests to run correctly. See other p3 salt formulas for the setup.

**Comment by lorengordon** [comment]
Umm. Wrong PR merged @ferricoxide? This one hadn't been re-reviewed or approved yet...

**Comment by ferricoxide** [comment]
> Umm. Wrong PR merged @ferricoxide? This one hadn't been re-reviewed or approved yet...

Goddammit. I was doing end-of-day cleanup tasks, saw greens and didn't notice that the review was missing. Guess the project wasn't (yet) configured to _require_ review before merging. :(

File: rsyslog/config/file.sls [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    - comment: |
+        ----------------------------------------
+        {{ loghost }}:
+          Neither the 'encrypted' nor the 'relp'
+          sub-options are currently supported by
+          this saltstack formula
+        ----------------------------------------
+    {%- elif values.protocol == 'udp' or values.protocol == 'tcp' %}
+file-{{ loghost }}:
+  file.managed:
```
[the remark this answers is not in the document]
Comment (ferricoxide) [reply rsyslog/config/file.sls:37]: Addressed in [BiffFixups](https://github.com/ferricoxide/rsyslog-formula/tree/BiffFixups) fork/branch

File: rsyslog/parameters/defaults.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    name: rsyslog
+  rootgroup: root
+  config: '/etc/rsyslog'
+  service:
+    name: rsyslog
+  subcomponent:
+    config: '/etc/rsyslog-subcomponent-formula.conf'
+  # Just here for testing
+  added_in_defaults: defaults_value
+  winner: defaults
```
[the remark this answers is not in the document]
Comment (ferricoxide) [reply rsyslog/parameters/defaults.yaml:17]: Addressed in [BiffFixups](https://github.com/ferricoxide/rsyslog-formula/tree/BiffFixups) fork/branch

File: rsyslog/files/default/example.tmpl.jinja
[the remark this answers is not in the document]
Comment (ferricoxide) [reply rsyslog/files/default/example.tmpl.jinja:1]: Addressed in [BiffFixups](https://github.com/ferricoxide/rsyslog-formula/tree/BiffFixups) fork/branch

File: rsyslog/files/default/example.tmpl
[the remark this answers is not in the document]
Comment (ferricoxide) [reply rsyslog/files/default/example.tmpl:1]: Addressed in [BiffFixups](https://github.com/ferricoxide/rsyslog-formula/tree/BiffFixups) fork/branch

File: .github/workflows/dependabot_hack.yml
[the remark this answers is not in the document]
Comment (ferricoxide) [reply .github/workflows/dependabot_hack.yml:1]: Addressed in [BiffFixups](https://github.com/ferricoxide/rsyslog-formula/tree/BiffFixups) fork/branch

**Pull request merged**